### PR TITLE
feat: added single videopage and static liked video page

### PR DESCRIPTION
**Pull request by mariyasada**
-added  static liked video page
-added requires auth for private routes
-added a single video page with functionality

https://user-images.githubusercontent.com/77275100/163849137-8f1ed2ed-a48b-44fe-9081-79d494fc382e.mp4

y

### CODE REVIEW

File: src/Components/LikedVideoCard/Likevideocard.css [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+  margin: 0.3rem 1rem -4rem 6rem;
+  width: 10rem;
+  justify-content: space-between;
+}
+.liked-video-icons {
+  font-size: 2rem;
+  justify-self: flex-end;
+  cursor: pointer;
+}
+@media only screen and (min-width: 320px) and (max-width: 880px) and (-webkit-min-device-pixel-ratio: 2) {
```
Comment (tusharkandpal) [inline src/Components/LikedVideoCard/Likevideocard.css:58]: liked you are using `min-width` and `max-width` to define proper range for media queries.

File: src/Components/Videocard/VideoCard.jsx [hunk 2]
```diff
@@ -22,12 +23,14 @@ export const VideoCard = ({ video }) => {
...
-          src={thumbnail}
-          alt={title}
-          onClick={() => navigate("/singlevideopage")}
-        />
+        <Link to={`/video/${_id}`}>
+          <img
+            className="image-of-video"
+            src={thumbnail}
+            alt={title}
+            // onClick={() => navigate("/singlevideopage")}
```
Comment (tusharkandpal) [inline src/Components/Videocard/VideoCard.jsx:31]: please remove this commented code.!

File: src/Components/LikedVideoCard/LikedVideoCard.jsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        <p className="liked-video-title">
+          How to make Resin Photo Frame ✨| step-by-step | tutorial | DIY
+          Anniversary Gift | Aara Vlogs
+        </p>
+        <p className="liked-video-channelname">Aesthetic Hues</p>
+        <div className="liked-video-icons-container flex-center">
+          <AiFillDislike className="liked-video-icons" />
+          <RiPlayList2Line className="liked-video-icons" />
+          <FaRegBookmark className="liked-video-icons" />
+          {/* <BsThreeDotsVertical className="liked-video-icons" /> */}
```
Comment (tusharkandpal) [inline src/Components/LikedVideoCard/LikedVideoCard.jsx:32]: please remove this, if no use further

File: src/Pages/LikedVideoPage/Likevideopage.css [hunk 4]
```diff
@@ -0,0 +1,33 @@
+.liked-video-with-sidebar-container {
+  height: auto;
+}
+.liked-videos-container {
+  margin: 10rem 0 2rem 20rem;
+  /* background-color: red; */
```
Comment (tusharkandpal) [inline src/Pages/LikedVideoPage/Likevideopage.css:6]: remove this.!

File: src/Pages/LikedVideoPage/Likevideopage.css [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+.liked-video-heading {
+  font-size: 2.3rem;
+  margin-left: 2rem;
+  color: var(--dark-stone);
+  align-self: flex-start;
+}
+@media only screen and (min-width: 320px) and (max-width: 880px) and (-webkit-min-device-pixel-ratio: 2) {
+  .videos-container {
+    flex-direction: column;
+    /* background-color: aqua; */
```
Comment (tusharkandpal) [inline src/Pages/LikedVideoPage/Likevideopage.css:23]: +1

File: src/Pages/NotFound/404error.jsx [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+        className="error-image-container"
+      />
+      <div className="errorpage-message flex-center ">
+        <span className="error-status">404 </span>
+        <p className="message-for-user">
+          Aha!!! You see ! You can be wrong ! or it could be us you should
+          probably{" "}
+          <Link to="/">
+            <span className="link-to-homepage">Go to HomePage</span>
+          </Link>
```
Comment (tusharkandpal) [inline src/Pages/NotFound/404error.jsx:20]: quite creative!

File: src/Pages/SingleVideoPage/SingleVideoPage.jsx [hunk 7]
```diff
@@ -1,10 +1,75 @@
...
+import { useParams } from "react-router-dom";
 
 export const SingleVideoPage = () => {
+  const { videos } = useVideos();
+  const { videoId } = useParams();
+
+  let singlevideo = videos.find((video) => video._id === videoId);
+  const { title, channelName, categoryName, avatarImage, date, views } =
+    singlevideo;
+  console.log(singlevideo);
```
Comment (tusharkandpal) [inline src/Pages/SingleVideoPage/SingleVideoPage.jsx:21]: +1

File: src/Pages/SingleVideoPage/singleVideopage.css [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+  padding: 5px;
+}
+.singlevideopage-title {
+  font-size: 2.2rem;
+  text-align: justify;
+  margin: 1rem;
+}
+.owner-category-date-container {
+  width: 30rem;
+  /* background-color: blue; */
```
Comment (tusharkandpal) [inline src/Pages/SingleVideoPage/singleVideopage.css:18]: please remove commented styles in this file.

File: src/Router/RequiresAuth.js [hunk 9]
```diff
@@ -0,0 +1,14 @@
+import React from 'react'
```
Comment (tusharkandpal) [inline src/Router/RequiresAuth.js:1]: this import is not mandatory as of higher versions of React ! can check & remove.

**Review by tusharkandpal — APPROVED** [review]
overall the code is clean & formatted.

Please deploy it asap.

File: src/Components/LikedVideoCard/LikedVideoCard.jsx [hunk 10]
```diff
@@ -0,0 +1,37 @@
+import React from "react";
+import "../LikedVideoCard/Likevideocard.css";
+import {
+  BsTrash,
+  RiPlayList2Line,
+  FaRegBookmark,
+  BsThreeDotsVertical,
+  AiFillDislike,
+} from "../Icons";
```
Comment (rkapoor10) [inline src/Components/LikedVideoCard/LikedVideoCard.jsx:9]: good practice

File: src/Pages/NotFound/404error.jsx [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+      <img
+        src="https://ik.imagekit.io/qrhnvir8bf0/business-3d-task_slyFJoiA1.png?ik-sdk-version=javascript-1.4.3&updatedAt=1650277790401"
+        alt="not-found"
+        className="error-image-container"
+      />
+      <div className="errorpage-message flex-center ">
+        <span className="error-status">404 </span>
+        <p className="message-for-user">
+          Aha!!! You see ! You can be wrong ! or it could be us you should
+          probably{" "}
```
Comment (rkapoor10) [inline src/Pages/NotFound/404error.jsx:17]: remove `{" "}`

File: src/Pages/SingleVideoPage/singleVideopage.css [hunk 12]
```diff
@@ -0,0 +1,91 @@
+.single-video-with-sidebar-container {
+  height: auto;
+  /* background-color: aqua; */
```
Comment (rkapoor10) [inline src/Pages/SingleVideoPage/singleVideopage.css:3]: remove comments if not needed

File: src/Pages/SingleVideoPage/singleVideopage.css [hunk 13]
```diff
@@ -0,0 +1,91 @@
...
+}
+.video-iframe-container {
+  margin: 10rem 0 2rem 20rem;
+  width: 74rem;
+  /* background-color: red; */
```
Comment (rkapoor10) [inline src/Pages/SingleVideoPage/singleVideopage.css:8]: remove comments if not needed

File: src/Pages/SingleVideoPage/singleVideopage.css [hunk 14]
```diff
@@ -0,0 +1,91 @@
...
+  .singlevideopage-title {
+    font-size: 1.6rem;
+  }
+  .channel-owner {
+    font-size: 1.3rem;
+  }
+  .cateogory-viwes-conatiner {
+    font-size: 10.5px;
+    gap: 1px;
+    /* background-color: aqua; */
```
Comment (rkapoor10) [inline src/Pages/SingleVideoPage/singleVideopage.css:78]: here also

**Review by rkapoor10 — APPROVED** [review]
- very well written CSS
 - folder structure is well maintained
 - auth implementation looks on point
 - clean up code a bit 
 overall code looks great...go-ahead for merge

**Pull request merged**